File: NirvanaSMS.Api.CSharp/Core/Base/CreditBase.cs
using NirvanaSMS.Api.Core.Interface;
using NirvanaSMS.Api.Core.Entity;
using NirvanaSMS.Api.Core.Utility;

namespace NirvanaSMS.Api.Core.Base
{
    public abstract class CreditBase : ICredit
    {
        public abstract ProcessResult<Credit> Query(ApiUser apiUser);
    }
}

File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/base/CreditBase.java
package org.nirvanasms.api.core.base;

import org.nirvanasms.api.core.entity.ApiUser;
import org.nirvanasms.api.core.entity.Credit;
import org.nirvanasms.api.core.interfaces.ICredit;
import org.nirvanasms.api.core.utility.ProcessResult;

public abstract class CreditBase implements ICredit {

	@Override
	public abstract ProcessResult<Credit> query(ApiUser apiUser);
}
